### PR TITLE
allow negative scale in PointwiseAffineTransform

### PR DESCRIPTION
**Pull request by mdmould**
The current implementation of PointwiseAffineTransform allows only positive values for the scale. As far as I can tell, this isn't necessary and only zero values (which make the transform no longer bijective) need to be rejected. Only the absolute value of the scale enters the log prob conversion while the scale and its reciprocal itself enters the transform of samples.

### CODE REVIEW

File: nflows/transforms/standard.py [hunk 1]
```diff
@@ -30,15 +30,15 @@ def __init__(
...
-        if not (scale > 0.0).all():
-            raise ValueError("Scale must be strictly positive.")
+        if (scale == 0.0).any():
+            raise ValueError("Scale must be non-zero.")
 
         self.register_buffer("_shift", shift)
         self.register_buffer("_scale", scale)
 
     @property
     def _log_scale(self) -> Tensor:
```
Comment (arturbekasov) [inline nflows/transforms/standard.py:40]: Can we rename this to `_log_abs_scale` to avoid confusion?

**Review by arturbekasov — CHANGES_REQUESTED** [review]
Can we also add some test cases with a negative scale to [AffineScalarTransformTest](https://github.com/bayesiains/nflows/blob/31a1b16013e49bf6943a7dfca9b45ebeb05cf5bd/tests/transforms/standard_test.py#L43)?

**Comment by arturbekasov** [comment]
Hey @mdmould (again),

Good idea. Left some comments on the implementation.

Cheers,

Artur

**Comment by imurray** [comment]
It's not clear to me that this change won't make some models fit worse than they did before. I'm not convinced it can be changed blindly (after all, if it made no difference, why bother?).

**Comment by arturbekasov** [comment]
> It's not clear to me that this change won't make some models fit worse than they did before. I'm not convinced it can be changed blindly (after all, if it made no difference, why bother?).

@imurray This is a non-trainable transformation with fixed parameters (note both `shift` and `scale` are buffers), so this will have no impact on model fits. The only effect is that you can now apply e.g. `PointwiseAffineTransform(shift=0, scale=-1)` (y = -x).

File: nflows/transforms/standard.py (hunk 1 above)
Comment (mdmould) [reply nflows/transforms/standard.py]: Done!

**Comment by mdmould** [comment]
I renamed to `_log_abs_scale` and added test with negative scales.

Cheers,
Matt

**Comment by mdmould** [comment]
I also added a test to check that scale=0 is caught.

**Comment by arturbekasov** [comment]
Thank you for taking the time, @mdmould, and sorry for the delay: merged!
